### PR TITLE
Moving the daemon to a Console class

### PR DESCRIPTION
**Pull request by nupplaphil**
Moving `processDaemon()` into a Console command.

### CODE REVIEW

**Comment by nupplaphil** [comment]
Tested with status/start/stop from console and daemon.php 

Works fine for me :)

@art4 - I did understand your wish for moving the process to the console rght, didn't I?

I will refactor the other Daemon and worker too, take this as a start :)

**Comment by Art4** [comment]
> @Art4 - I did understand your wish for moving the process to the console rght, didn't I?
> 
> I will refactor the other Daemon and worker too, take this as a start :)

That's correct, thank you very much. :+1:

**Review by Art4 — COMMENTED** [review]
I tested the commands without issues:

```bash
bin/daemon.php start
bin/daemon.php status
bin/daemon.php stop
bin/console.php daemon start
bin/console.php daemon status
bin/console.php daemon stop
```

Should we add a note in `bin/daemon.php` about the `bin/console.php daemon` command or can we even deprecate the `bin/daemon.php` file?

**Comment by MrPetovan** [comment]
Many current installs will need to manually change their setup once the console command is created, so we will need to maintain the `bin/daemon.php` endpoint for at least a full release.

File: src/Core/Console.php [hunk 1]
```diff
@@ -37,6 +37,7 @@ protected function getHelp()
 	config                 Edit site config
 	contact                Contact management
 	createdoxygen          Generate Doxygen headers
+	daemon                 Interacting with the Friendica daemons
```
Comment (MrPetovan) [inline src/Core/Console.php]: I don't believe we are breeding daemons at Friendica, we only have that trusty one.
```suggestion
	daemon                 Interact with the Friendica daemon
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Core/Console.php (hunk 1 above)
Comment (nupplaphil) [reply src/Core/Console.php]: good catch .. I thought about moving jetstream to this command as well - so one daemon - many daemons, isn't it? hahaha

**Review by MrPetovan — APPROVED** [review]

**Comment by Art4** [comment]
> Many current installs will need to manually change their setup once the console command is created, so we will need to maintain the `bin/daemon.php` endpoint for at least a full release.

Of course, thats why it would be good to have note or deprecation to notify the users.

**Comment by MrPetovan** [comment]
Ah, yes, this should be done in the release notes when we release version 2025.02, ping @tobiasd . I guess this can also be added in the code whenever `bin/daemon.php` is called.

**Pull request merged**